src/IFrameNavigator.ts

```diff
@@ -156,27 +172,15 @@ export default class IFrameNavigator implements Navigator {
     private newElementId: string | null;
     private isLoading: boolean;
 
-    public static async create(
-        element: HTMLElement,
-        manifestUrl: URL,
-        store: Store,
-        cacher: Cacher,
-        settings: BookSettings,
-        annotator: Annotator | null = null,
-        paginator: PaginatedBookView | null = null,
-        scroller: ScrollingBookView | null = null,
-        eventHandler: EventHandler | null = null,
-        upUrl: URL | null = null,
-        upLabel: string | null = "Back"
-        ) {
-
+    public static async create(config: IFrameNavigatorConfig) {
         const navigator = new this(
-            store, cacher, settings, annotator,
-            paginator, scroller, eventHandler,
-            upUrl, upLabel
+            config.store, config.cacher, config.settings, config.annotator || null,
+            config.paginator || null, config.scroller || null, config.eventHandler || null,
+            config.upLink ? config.upLink.url || null : null,
+            config.upLink ? config.upLink.label || null : null
```

Comment:
Ow. This is not very pretty but no suggestions immediately jump out at me as being less painful. Blame JavaScript for having `null` _and_ `undefined`, I suppose.